integrations/integration_test.go

```diff
@@ -130,6 +132,18 @@ func initIntegrationTest() {
 		if _, err = db.Exec("CREATE DATABASE testgitea"); err != nil {
 			log.Fatalf("db.Exec: %v", err)
 		}
+	case setting.UseMSSQL:
+		host, port := models.ParseMSSQLHostPort(models.DbCfg.Host)
+		db, err := sql.Open("mssql", fmt.Sprintf("server=%s; port=%s; database=%s; user id=%s; password=%s;",
+			host, port, "master", models.DbCfg.User, models.DbCfg.Passwd))
+		defer db.Close()
```

Comment:
The `defer` should be after the err check because db maybe nil before that. And the above databases code also have the problem.

Reply:
I've changed that, I hope its what you ment.